Neos.Neos/Classes/Domain/NodeLabel/NodeLabelRenderer.php

```diff
@@ -2,24 +2,40 @@
 
 declare(strict_types=1);
 
-namespace Neos\ContentRepositoryRegistry\Factory\NodeTypeManager;
+namespace Neos\Neos\Domain\NodeLabel;
 
-use Neos\ContentRepository\Core\NodeType\NodeLabelGeneratorFactoryInterface;
-use Neos\ContentRepository\Core\NodeType\NodeLabelGeneratorInterface;
 use Neos\ContentRepository\Core\NodeType\NodeType;
-use Neos\ContentRepositoryRegistry\NodeLabel\ExpressionBasedNodeLabelGenerator;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
 use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Neos\Fusion\Helper\NodeHelper;
+use Neos\Neos\Utility\NodeTypeWithFallbackProvider;
 
-class ObjectManagerBasedNodeLabelGeneratorFactory implements NodeLabelGeneratorFactoryInterface
+/**
+ * @api For PHP, in Fusion one can use ${Neos.Node.renderLabel(node)} {@see NodeHelper::renderLabel()}
+ */
+final readonly class NodeLabelRenderer
 {
+    use NodeTypeWithFallbackProvider;
+
     public function __construct(
-        private readonly ObjectManagerInterface $objectManager
+        private ContentRepositoryRegistry $contentRepositoryRegistry,
+        private ObjectManagerInterface $objectManager
     ) {
     }
 
-    public function create(NodeType $nodeType): NodeLabelGeneratorInterface
+    public function renderNodeLabel(Node $node): string
     {
+        $nodeType = $this->getNodeType($node);
+        $generator = $this->getNodeLabelGeneratorForNodeType($nodeType);
+        return $generator->getLabel($node);
+    }
 
+    /**
+     * @internal
+     */
+    public function getNodeLabelGeneratorForNodeType(NodeType $nodeType): NodeLabelGeneratorInterface
```

Comment:
idk thought it would be kinda legitimate (as in a internal edgecase way and for testing) to get an instance of a `NodeLabelGeneratorInterface` as that thing is api since ages and `$nodeType->getNodeLabelGenerator()` was previously also api.

Reply:
"You Aren't Gonna Need It" had to look up that one 😂 youre to good with acronyms - but i should be the hipster.

... thats why `@internal`. Ill request another opinion and summon the one and only @kitsunet may he decide.

Reply:
oops, sorry for that – it used to be a very common acronym because it was *very common to add code for a "maybe we need this at some point"-case* that never occurred.

Sorry to be stubborn but I don't see a single reason why we should make it public *now*. And I'm sure you agree that the `@internal` annotation is just a hack because we couldn't find a better abstraction.

Let's rather continue our journey to make `Node` easier to mock/create (which this PR is a great step towards) so we can test the actual API

Reply:
The way i see it we dont have to be more catholic than the pope regarding api that is in Neos.Neos. Its anticorruption around the ESCR :D 😂 And in 8.3 people could get this `NodeLabelGeneratorInterface` and why not make it at least possible without reflection (and same for allowing us to build unit tests). Sure we could use a factory pattern here but why complex if Flows object framework supports and even encourages simplicity. In the end i will not insist to much on my opinion... lets see if someone wants to add their two cents.

Reply:
Yes ill remove it then as discussed also with @kitsunet